Pair EDA

In [13]:
import pandas as pd
import numpy as np

Para realizar estos ejercicios deberéis usar el conjunto de datos de world-data-2023-part1.csv.

En este fichero encontraremos la siguiente información:

Country: Nombre del país.

Density\n(P/Km2): Densidad de población medida en personas por kilómetro cuadrado.

Abbreviation: Abreviatura o código que representa al país.

Agricultural Land( %): Porcentaje del área total de tierra utilizada con fines agrícolas.

Land Area(Km2): Área total de tierra del país en kilómetros cuadrados.

Armed Forces size: Tamaño de las fuerzas armadas en el país.

Birth Rate: Número de nacimientos por cada 1,000 habitantes por año.

Calling Code: Código internacional de llamada para el país.

Capital/Major City: Nombre de la capital o ciudad principal.

Co2-Emissions: Emisiones de dióxido de carbono en toneladas.

CPI: Índice de Precios al Consumidor (IPC), una medida de la inflación y el poder adquisitivo.

CPI Change (%): Cambio porcentual en el Índice de Precios al Consumidor en comparación con el año anterior.

Currency-Code: Código de moneda utilizado en el país.

Fertility Rate: Número promedio de hijos nacidos de una mujer durante su vida.

Forested Area (%): Porcentaje del área total cubierta por bosques.

Gasoline Price: Precio de la gasolina por litro en moneda local.

Ejercicios de exploración del conjunto de datos

In [14]:
df_world_data = pd.read_csv('world-data-2023_part1.csv', index_col=0)

Muestra las primeras 5 filas del conjunto de datos.

In [15]:
df_world_data.head()

,Country,Density\n(P/Km2),Abbreviation,Agricultural Land( %),Land Area(Km2),Armed Forces size,Birth Rate,Calling Code,Capital/Major City,Co2-Emissions,CPI,CPI Change (%),Currency-Code,Fertility Rate,Forested Area (%),Gasoline Price
0,Afghanistan,60,AF,58.10%,"652,230","323,000",32.49,93.0,Kabul,"8,672",149.9,2.30%,AFN,4.47,2.10%,$0.70
1,Albania,105,AL,43.10%,"28,748","9,000",11.78,355.0,Tirana,"4,536",119.05,1.40%,ALL,1.62,28.10%,$1.36
2,Algeria,18,DZ,17.40%,"2,381,741","317,000",24.28,213.0,Algiers,"150,006",151.36,2.00%,DZD,3.02,0.80%,$0.28
3,Andorra,164,AD,40.00%,468,NaN,7.20,376.0,Andorra la Vella,469,NaN,NaN,EUR,1.27,34.00%,$1.51
4,Angola,26,AO,47.50%,"1,246,700","117,000",40.73,244.0,Luanda,"34,693",261.73,17.10%,AOA,5.52,46.30%,$0.97


Muestra las últimas 5 filas del conjunto de datos.

In [16]:
df_world_data.tail()

,Country,Density\n(P/Km2),Abbreviation,Agricultural Land( %),Land Area(Km2),Armed Forces size,Birth Rate,Calling Code,Capital/Major City,Co2-Emissions,CPI,CPI Change (%),Currency-Code,Fertility Rate,Forested Area (%),Gasoline Price
190,Venezuela,32,VE,24.50%,"912,050","343,000",17.88,58.0,Caracas,"164,175","2,740.27",254.90%,VED,2.27,52.70%,$0.00
191,Vietnam,314,VN,39.30%,"331,210","522,000",16.75,84.0,Hanoi,"192,668",163.52,2.80%,VND,2.05,48.10%,$0.80
192,Yemen,56,YE,44.60%,"527,968","40,000",30.45,967.0,Sanaa,"10,609",157.58,8.10%,YER,3.79,1.00%,$0.92
193,Zambia,25,ZM,32.10%,"752,618","16,000",36.19,260.0,Lusaka,"5,141",212.31,9.20%,ZMW,4.63,65.20%,$1.40
194,Zimbabwe,38,ZW,41.90%,"390,757","51,000",30.68,263.0,Harare,"10,983",105.51,0.90%,NaN,3.62,35.50%,$1.34


Muestra una muestra aleatoria de 5 filas del conjunto de datos.

In [17]:
df_world_data.sample(5)

,Country,Density\n(P/Km2),Abbreviation,Agricultural Land( %),Land Area(Km2),Armed Forces size,Birth Rate,Calling Code,Capital/Major City,Co2-Emissions,CPI,CPI Change (%),Currency-Code,Fertility Rate,Forested Area (%),Gasoline Price
80,Iraq,93,IQ,21.40%,"438,317","209,000",29.08,964.0,Baghdad,"190,061",119.86,0.40%,IQD,3.67,1.90%,$0.61
58,Fiji,49,FJ,23.30%,"18,274","4,000",21.28,679.0,Suva,"2,046",132.3,1.80%,FJD,2.77,55.90%,$0.82
159,Solomon Islands,25,SB,3.90%,"28,896",NaN,32.44,677.0,Honiara,169,133.06,1.60%,SBD,4.40,77.90%,NaN
111,Federated States of Micronesia,784,FM,31.40%,702,NaN,22.82,691.0,Palikir,143,112.1,0.50%,USD,3.05,91.90%,NaN
73,Vatican City,"2,003",NaN,NaN,0,NaN,NaN,379.0,Vatican City,NaN,NaN,NaN,EUR,NaN,NaN,NaN


Muestra el nombre de todas las columnas en el conjunto de datos.

In [18]:
df_world_data.columns

Index(['Country', 'Density\n(P/Km2)', 'Abbreviation', 'Agricultural Land( %)',
       'Land Area(Km2)', 'Armed Forces size', 'Birth Rate', 'Calling Code',
       'Capital/Major City', 'Co2-Emissions', 'CPI', 'CPI Change (%)',
       'Currency-Code', 'Fertility Rate', 'Forested Area (%)',
       'Gasoline Price'],
      dtype='object')

Muestra los valores únicos en la columna "Abbreviation".

In [20]:
df_world_data['Abbreviation'].unique()

array(['AF', 'AL', 'DZ', 'AD', 'AO', 'AG', 'AR', 'AM', 'AU', 'AT', 'AZ',
       'BS', 'BH', 'BD', 'BB', 'BY', 'BE', 'BZ', 'BJ', 'BT', 'BO', 'BA',
       'BW', 'BR', 'BN', 'BG', 'BF', 'BI', 'CI', 'CV', 'KH', 'CM', 'CA',
       'CF', 'TD', 'CL', 'CN', 'CO', 'KM', nan, 'CR', 'HR', 'CU', 'CY',
       'CZ', 'CD', 'DK', 'DJ', 'DM', 'DO', 'EC', 'EG', 'SV', 'GQ', 'ER',
       'EE', 'ET', 'FJ', 'FI', 'FR', 'GA', 'GM', 'GE', 'DE', 'GH', 'GR',
       'GD', 'GT', 'GN', 'GW', 'GY', 'HT', 'HN', 'HU', 'IS', 'IN', 'ID',
       'IR', 'IQ', 'IL', 'IT', 'JM', 'JP', 'JO', 'KZ', 'KE', 'KI', 'KW',
       'KG', 'LA', 'LV', 'LB', 'LS', 'LR', 'LY', 'LI', 'LT', 'LU', 'MG',
       'MW', 'MY', 'MV', 'ML', 'MT', 'MH', 'MR', 'MU', 'MX', 'FM', 'MD',
       'MC', 'MN', 'ME', 'MA', 'MZ', 'MM', 'NR', 'NP', 'NL', 'NZ', 'NI',
       'NE', 'NG', 'KP', 'NO', 'OM', 'PK', 'PW', 'PA', 'PG', 'PY', 'PE',
       'PH', 'PL', 'PT', 'QA', 'RO', 'RU', 'RW', 'KN', 'LC', 'VC', 'WS',
       'SM', 'ST', 'SA', 'SN', 'RS', 'SC', 'SL', 'SG

In [21]:
df_world_data['Abbreviation'].nunique()

188

In [ ]:
df_world_data['Abbreviation'].value_counts() # conteo valores unicos

Abbreviation
AF    1
AL    1
DZ    1
AD    1
AO    1
     ..
VE    1
VN    1
YE    1
ZM    1
ZW    1
Name: count, Length: 188, dtype: int64

Cuenta cuántas veces aparece cada valor único en la columna "Abbreviation".

In [ ]:
df_world_data['Abbreviation'].size ## numero total de elementos

195

Verifica si hay valores nulos en el conjunto de datos.

In [28]:
df_world_data['Abbreviation'].isna().sum()

np.int64(7)

Verifica si hay valores no nulos en el conjunto de datos.

In [29]:
df_world_data['Abbreviation'].notnull().sum()

np.int64(188)

Muestra información general sobre el conjunto de datos.

In [32]:
df_world_data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 195 entries, 0 to 194
Data columns (total 16 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Country                195 non-null    object 
 1   Density
(P/Km2)        195 non-null    object 
 2   Abbreviation           188 non-null    object 
 3   Agricultural Land( %)  188 non-null    object 
 4   Land Area(Km2)         194 non-null    object 
 5   Armed Forces size      171 non-null    object 
 6   Birth Rate             189 non-null    float64
 7   Calling Code           194 non-null    float64
 8   Capital/Major City     192 non-null    object 
 9   Co2-Emissions          188 non-null    object 
 10  CPI                    178 non-null    object 
 11  CPI Change (%)         179 non-null    object 
 12  Currency-Code          180 non-null    object 
 13  Fertility Rate         188 non-null    float64
 14  Forested Area (%)      188 non-null    object 
 15  Gasoline Pr

Dimensiones y Filas
Index: 195 entries, 0 to 194: Esto indica que el DataFrame tiene 195 filas (o "entradas"). El índice de las filas va desde el valor 0 hasta el valor 194.

Data columns (total 16 columns):: Esto indica que el DataFrame contiene un total de 16 columnas

In [ ]:
Conteo de Valores No Nulos ($Non-Null Count$) Permite detectar valores faltantes. Si este número es menor que el número total de filas, hay datos ausentes en esa columna.

In [ ]:
df_world_data.describe() ## obtenemos info con valores estadisticos(solo nos da info numerica)

,Birth Rate,Calling Code,Fertility Rate
count,189.000000,194.000000,188.000000
mean,20.214974,360.546392,2.698138
std,9.945774,323.236419,1.282267
min,5.900000,1.000000,0.980000
25%,11.300000,82.500000,1.705000
50%,17.950000,255.500000,2.245000
75%,28.750000,506.750000,3.597500
max,46.080000,1876.000000,6.910000


Elimina una columna específica (por ejemplo, "Armed Forces size").

In [ ]:
df_world_data.drop(columns='Fertility Rate') ##no se guarda en la memoria

,Country,Density\n(P/Km2),Abbreviation,Agricultural Land( %),Land Area(Km2),Armed Forces size,Birth Rate,Calling Code,Capital/Major City,Co2-Emissions,CPI,CPI Change (%),Currency-Code,Forested Area (%),Gasoline Price
0,Afghanistan,60,AF,58.10%,"652,230","323,000",32.49,93.0,Kabul,"8,672",149.9,2.30%,AFN,2.10%,$0.70
1,Albania,105,AL,43.10%,"28,748","9,000",11.78,355.0,Tirana,"4,536",119.05,1.40%,ALL,28.10%,$1.36
2,Algeria,18,DZ,17.40%,"2,381,741","317,000",24.28,213.0,Algiers,"150,006",151.36,2.00%,DZD,0.80%,$0.28
3,Andorra,164,AD,40.00%,468,NaN,7.20,376.0,Andorra la Vella,469,NaN,NaN,EUR,34.00%,$1.51
4,Angola,26,AO,47.50%,"1,246,700","117,000",40.73,244.0,Luanda,"34,693",261.73,17.10%,AOA,46.30%,$0.97
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
190,Venezuela,32,VE,24.50%,"912,050","343,000",17.88,58.0,Caracas,"164,175","2,740.27",254.90%,VED,52.70%,$0.00
191,Vietnam,314,VN,39.30%,"331,210","522,000",16.75,84.0,Hanoi,"192,668",163.52,2.80%,VND,48.10%,$0.80
192,Yemen,56,YE,44.60%,"527,968","40,000",30.45,967.0,Sanaa,"10,609",157.58,8.10%,YER,1.00%,$0.92
193,Zambia,25,ZM,32.10%,"752,618","16,000",36.19,260.0,Lusaka,"5,141",212.31,9.20%,ZMW,65.20%,$1.40


In [ ]:
df_world_data.drop(columns='Fertility Rate', inplace=True)  #guardo los cambios (sale error xq he ejecutado dos veces)

KeyError: "['Fertility Rate'] not found in axis"

In [38]:
df_world_data

,Country,Density\n(P/Km2),Abbreviation,Agricultural Land( %),Land Area(Km2),Armed Forces size,Birth Rate,Calling Code,Capital/Major City,Co2-Emissions,CPI,CPI Change (%),Currency-Code,Forested Area (%),Gasoline Price
0,Afghanistan,60,AF,58.10%,"652,230","323,000",32.49,93.0,Kabul,"8,672",149.9,2.30%,AFN,2.10%,$0.70
1,Albania,105,AL,43.10%,"28,748","9,000",11.78,355.0,Tirana,"4,536",119.05,1.40%,ALL,28.10%,$1.36
2,Algeria,18,DZ,17.40%,"2,381,741","317,000",24.28,213.0,Algiers,"150,006",151.36,2.00%,DZD,0.80%,$0.28
3,Andorra,164,AD,40.00%,468,NaN,7.20,376.0,Andorra la Vella,469,NaN,NaN,EUR,34.00%,$1.51
4,Angola,26,AO,47.50%,"1,246,700","117,000",40.73,244.0,Luanda,"34,693",261.73,17.10%,AOA,46.30%,$0.97
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
190,Venezuela,32,VE,24.50%,"912,050","343,000",17.88,58.0,Caracas,"164,175","2,740.27",254.90%,VED,52.70%,$0.00
191,Vietnam,314,VN,39.30%,"331,210","522,000",16.75,84.0,Hanoi,"192,668",163.52,2.80%,VND,48.10%,$0.80
192,Yemen,56,YE,44.60%,"527,968","40,000",30.45,967.0,Sanaa,"10,609",157.58,8.10%,YER,1.00%,$0.92
193,Zambia,25,ZM,32.10%,"752,618","16,000",36.19,260.0,Lusaka,"5,141",212.31,9.20%,ZMW,65.20%,$1.40


Selecciona las columnas numéricas del conjunto de datos y muestra sus estadísticas descriptivas básicas. para esto tendréis que usar el método '.describe().

In [40]:
df_world_data.select_dtypes(include=['int','float'])

,Birth Rate,Calling Code
0,32.49,93.0
1,11.78,355.0
2,24.28,213.0
3,7.20,376.0
4,40.73,244.0
...,...,...
190,17.88,58.0
191,16.75,84.0
192,30.45,967.0
193,36.19,260.0


In [41]:
df_world_data.select_dtypes(include=['int','float']).describe()

,Birth Rate,Calling Code
count,189.000000,194.000000
mean,20.214974,360.546392
std,9.945774,323.236419
min,5.900000,1.000000
25%,11.300000,82.500000
50%,17.950000,255.500000
75%,28.750000,506.750000
max,46.080000,1876.000000


In [42]:
df_world_data.select_dtypes(include=['int','float']).describe().T

,count,mean,std,min,25%,50%,75%,max
Birth Rate,189.0,20.214974,9.945774,5.9,11.3,17.95,28.75,46.08
Calling Code,194.0,360.546392,323.236419,1.0,82.5,255.50,506.75,1876.00
